solidity/contracts/ChainlinkLib.sol

```diff
@@ -12,9 +12,39 @@ library ChainlinkLib {
     bytes32 jobId;
     address callbackAddress;
     bytes4 callbackFunctionId;
+    bytes32 externalId;
     Buffer.buffer buf;
   }
 
+  function initialize(
+    Run memory self,
+    bytes32 _jobId,
+    address _callbackAddress,
+    string _callbackFunctionSignature
+  ) internal pure returns (ChainlinkLib.Run memory) {
+    Buffer.init(self.buf, 128);
+    self.jobId = _jobId;
+    self.callbackAddress = _callbackAddress;
+    self.callbackFunctionId = bytes4(keccak256(_callbackFunctionSignature));
+    self.buf.startMap();
+    return self;
+  }
+
+  function encodeForOracle(
+    Run memory self,
+    bytes4 _oracleFid,
+    uint256 _clArgsVersion
+  ) internal pure returns (bytes memory) {
+    return abi.encodeWithSelector(
```

Comment:
Looks like this method will require the caller to call `close()` before invoking. It's also the only oracle related logic in this file.